### PR TITLE
[litmus] Fix wrong behavior in presi mode

### PR DESCRIPTION
**Pull request by maranget**
Combination of options `-mode presi -ascall false` was producing C code that could not compile when there is no exported labels in code.

**Notice:** If some labels need to be exported legitimately, then the setting `-ascall true` applies without notice.

### CODE REVIEW

**Comment by maranget** [comment]
Problem found with PR #845.

File: litmus/preSi.ml [hunk 1]
```diff
@@ -98,7 +98,7 @@ module Make
       Cfg.ascall || Cfg.is_kvm || do_label_init || CfgLoc.need_prelude
       || Cfg.variant Variant_litmus.Self
 
-    let do_precise = Fault.Handling.is_fatal Cfg.precision
+    let do_precise = Cfg.is_kvm && Fault.Handling.is_fatal Cfg.precision
```
[the remark this answers is not in the document]
Comment (maranget) [reply litmus/preSi.ml:101]: Tests triggered by `make litmus-cata-aarch64-test-presi` were all failling. The other changes in the PR are here for robustness.

[the remark this answers is not in the document]
Comment (relokin) [reply litmus/preSi.ml:101]: ah now I get it, forcing github to ignore whitespace changes made it more clear, you've only enclosed a block in a begin ... end.

**Review by relokin — APPROVED** [review]

File: litmus/preSi.ml (hunk 1 above)
[the remark this answers is not in the document]
Comment (maranget) [reply litmus/preSi.ml:101]: Exactly, I should probably configure github similarly.

**Comment by maranget** [comment]
Merged, thanks @relokin.

**Pull request merged**